significant_strikes.py

```diff
@@ -0,0 +1,500 @@
+        rounds = ['R1', 'R2', 'R3', 'R4', 'R5', 'overall']
+        time_periods = ['last-3-fights', 'last-5-fights', 'alltime']
+
+        # Iterate through all combinations to create column names
+        for fighter in fighters:
+            for significant_strike_stat in significant_strike_stats:
+                for round in rounds:
+                    for time_period in time_periods:
+                        col_name = f"{fighter.replace(' ', '_')}_{significant_strike_stat.replace(' ', '_')}_{round}_{time_period.replace(' ', '_')}"
+                        col_names.append(col_name)
+        return col_names
+
+    def create_col_names_differential_significant_strikes(self):
+        """
+        Generates column names for differential significant strike statistics.
+
+        Returns:
+            list: A list of strings, each representing a column name for a specific significant strike statistic.
+        """
+
+        col_names = []
+        fighters = ['fighter-a', 'fighter-b']
+        significant_strike_stats = ['significant-strikes-landed-differential', 'significant-strikes-accuracy-differential', 'significant-strikes-defense-differential', 'significant-strikes-absorbed-differential']
+        rounds = ['R1', 'R2', 'R3', 'R4', 'R5', 'overall']
+        time_periods = ['last-3-fights', 'last-5-fights', 'alltime']
+
+        # Iterate through all combinations to create column names
+        for fighter in fighters:
+            for significant_strike_stat in significant_strike_stats:
+                for round in rounds:
+                    for time_period in time_periods:
+                        col_name = f"{fighter.replace(' ', '_')}_{significant_strike_stat.replace(' ', '_')}_{round}_{time_period.replace(' ', '_')}"
+                        col_names.append(col_name)
+        return col_names
+    
+    def get_fighter_fights_significant_strikes(self, df, fighter_id, index):
+        """
+        Retrieves the most recent fights of a fighter before a given index in the dataframe.
+
+        Args:
+            df (pd.DataFrame): The dataframe containing fight records.
+            fighter_id (str): The ID of the fighter.
+            index (int): The index in the dataframe up to which fights should be considered.
+
+        Returns:
+            tuple: Contains three dataframes - the last 3 fights, the last 5 fights, and all previous fights.
+        """
+
+        all_prev_fights = df.loc[:index-1]
+
+        if not all_prev_fights.empty:
+            # Filter fights involving the specified fighter
+            fighter_a_id_vals = all_prev_fights.fighter_a_id.values
+            fighter_b_id_vals = all_prev_fights.fighter_b_id.values
+
+            prev_fights = all_prev_fights[((fighter_a_id_vals == fighter_id) | (fighter_b_id_vals == fighter_id))]
+            num_fights = len(prev_fights)
```

Comment:
A more concise way to do this would be:

`return prev_fights[-3:], prev_fights[-5:], prev_fights`

This gets the last X fights as required but also gets the entire dataset if there are less than X rows.